File: services/shop-backend/src/main/java/io/miragon/shop/application/port/outbound/CartRepository.java
package io.miragon.shop.application.port.outbound;

import io.miragon.shop.domain.cart.Cart;
import io.miragon.shop.domain.shared.UserId;

public interface CartRepository {
    Cart loadCart(UserId userId);
    void storeCart(Cart cart);
    void deleteCart(UserId userId);
}

File: services/shop/shop-backend/src/main/kotlin/io/miragon/shop/application/port/outbound/CartRepository.kt
package io.miragon.shop.application.port.outbound

import io.miragon.shop.domain.cart.Cart
import io.miragon.shop.domain.shared.UserId

interface CartRepository {
    fun loadCart(userId: UserId): Cart
    fun storeCart(cart: Cart)
    fun deleteCart(userId: UserId)
}